src/Microsoft.Azure.WebJobs.Protocols/FunctionDescriptor.cs

```diff
@@ -35,5 +36,31 @@ public class FunctionDescriptor
         /// </summary>
         [JsonIgnore]
         internal MethodInfo Method { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="TriggerParameterDescriptor"/> for this function
+        /// </summary>
+        [JsonIgnore]
+        internal TriggerParameterDescriptor TriggerDescriptor { get; set; }
+#if PUBLICPROTOCOL
+#else
```

Comment:
1. The public protocol is used by the dashboard.  These code files are shared between Webjobs.Host and Dashboard. 

2. However, this reflection data isn't available or necessary on the dashboard side, only for host.